### PR TITLE
Multiple disks/volumes for storing data for send in Distributed engine

### PR DESCRIPTION
**Pull request by azat**
Changelog category (leave one):
- New Feature

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Multiple disks/volumes for storing data for send in Distributed engine

Detailed description (optional):

Usage example:

```sql
CREATE TABLE foo (key Int) Engine=Distributed(test_shard_localhost, currentDatabase(), some_table, key%2, 'default');
```

Follow-up: #4918

v2: use `storage_policy` over introducing separate `distributed_storage_policy`

### CODE REVIEW

File: dbms/src/Storages/StorageDistributed.cpp [hunk 1]
```diff
@@ -481,57 +529,45 @@ bool StorageDistributed::hasColumn(const String & column_name) const
...
 
 
-void StorageDistributed::requireDirectoryMonitor(const std::string & name)
+void StorageDistributed::requireDirectoryMonitor(const std::string & disk, const std::string & name)
 {
-    std::lock_guard lock(cluster_nodes_mutex);
-    cluster_nodes_data[name].requireDirectoryMonitor(name, *this, monitors_blocker);
-}
+    const std::string path(disk + relative_data_path + name);
+    LOG_DEBUG(log, "Creating directory monitor for " << path);
```
Comment (azat) [inline dbms/src/Storages/StorageDistributed.cpp]: This will contain user:pass, not sure that this is a good idea

Comment (alexey-milovidov) [reply dbms/src/Storages/StorageDistributed.cpp]: Yes, better to remove.

**Comment by azat** [comment]
#8750 had been merged, this PR rebased against master (WIP dropped)

**Comment by azat** [comment]
- PVS check - Unable to start the analysis on this file.
- Integration tests -- test_cluster_copier (does not looks related to this change)

**Comment by azat** [comment]
>Integration tests -- test_cluster_copier (does not looks related to this change)

Even though I cannot reproduce it when running manually, it fails constantly on CI, digging into logs

**Comment by azat** [comment]
>2020.01.25 17:51:31.841341 [ 1 ] {} <Error> ClusterCopier: Will retry: Code: 49, e.displayText() = DB::Exception: Disk path must ends with '/', but '/var/log/clickhouse-server/copier/clickhouse-copier_20200125175131_66' doesn't., Stack trace (when copying this message, always include the lines below):

[Fixed](https://github.com/ClickHouse/ClickHouse/pull/8756/files#diff-2b642ee0c4ebb5478dbfc4f4eeae19baR275)

*(The reason it passed in my setup is due to stalled build objects)*

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**